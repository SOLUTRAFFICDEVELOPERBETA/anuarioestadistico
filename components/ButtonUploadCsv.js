import React from 'react';
import PropTypes from 'prop-types';
import { Box, Button } from '@material-ui/core';
import { CloudUpload } from '@material-ui/icons';

/**
 * @description Componente que permite la descarga de un csv
 * @param {String}  title titulo del componente,
 * @param {func: ()=> void} handleFilesAnnex función para subir el archivo,
 * @param {String} name Nombre deñ archivo,
 * @param {string} id  string identificador  del campo,
 * @param {String} htmlFor Identificador del Label,
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const ButtonUploadCsv = ({ title, handleFilesAnnex, name, id, htmlFor }) => {
    return (
        <Box>
            <input
                type={'file'}
                name={name}
                id={id}
                multiple={false}
                style={{ display: 'none' }}
                onChange={(e) => handleFilesAnnex(e.target.files)}
            />
            <label htmlFor={htmlFor}>
                <Button
                    endIcon={<CloudUpload />}
                    component={'div'}
                    size={'small'}
                    fullWidth={true}
                    variant={'contained'}
                    color={'secondary'}>
                    {title}
                </Button>
            </label>
        </Box>
    );
};
ButtonUploadCsv.propTypes = {
    title: PropTypes.string,
    handleFilesAnnex: PropTypes.func,
    name: PropTypes.string,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    htmlFor: PropTypes.string
};
export default ButtonUploadCsv;
