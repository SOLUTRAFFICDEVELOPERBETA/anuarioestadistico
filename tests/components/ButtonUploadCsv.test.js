import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import ButtonUploadCsv from '../../components/ButtonUploadCsv';

describe('Pruebas en <ButtonUploadCsv/>', () => {
    const handleFilesAnnex = jest.fn();
    const wrapper = shallow(
        <ButtonUploadCsv
            title="Titulo de prueba"
            handleFilesAnnex={handleFilesAnnex}
            name="img-pruebas"
            id={123}
            htmlFor={'mg-pruebas'}
        />
    );
    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();


    })

    test('debe llamar la funciÃ³n handleFilesAnnex con los argumentos enviados ', () => {
        const parametros = {
            target: {
                files: [
                    {
                        file: 'https://www.antiagingya.com/es/wp-content/uploads/2015/01/img-default-autores.jpg',
                        name: 'imgDefault',
                        size: 343565667,
                        type: 'img',
                    }
                ]
            }
        }
        wrapper.find('input').simulate('change', parametros);
        expect(handleFilesAnnex).toHaveBeenCalledTimes(1);
        expect(handleFilesAnnex).toHaveBeenCalledWith(expect.any(Object));
        expect(handleFilesAnnex).toHaveBeenCalledWith(parametros.target.files);


    })
})