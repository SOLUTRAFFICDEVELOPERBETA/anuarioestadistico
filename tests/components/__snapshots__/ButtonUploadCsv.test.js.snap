// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ButtonUploadCsv/> debe mostrarse correctamente 1`] = `
<Styled(MuiBox)>
  <input
    id={123}
    multiple={false}
    name="img-pruebas"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <label
    htmlFor="mg-pruebas"
  >
    <WithStyles(ForwardRef(Button))
      color="secondary"
      component="div"
      endIcon={<Memo(CloudUploadIcon) />}
      fullWidth={true}
      size="small"
      variant="contained"
    >
      Titulo de prueba
    </WithStyles(ForwardRef(Button))>
  </label>
</Styled(MuiBox)>
`;
